runtime/executor/src/main/java/org/apache/nemo/runtime/executor/task/TaskExecutor.java

```diff
@@ -213,13 +213,18 @@ public TaskExecutor(final Task task,
 
       // Additional outputs
       final Map<String, List<NextIntraTaskOperatorInfo>> internalAdditionalOutputMap =
-        getInternalAdditionalOutputMap(irVertex, irVertexDag, edgeIndexMap, operatorWatermarkManagerMap);
+        getInternalOutputMap(irVertex, irVertexDag, edgeIndexMap, operatorWatermarkManagerMap);
       final Map<String, List<OutputWriter>> externalAdditionalOutputMap =
         getExternalAdditionalOutputMap(irVertex, task.getTaskOutgoingEdges(), intermediateDataIOFactory);
 
       // Main outputs
-      final List<NextIntraTaskOperatorInfo> internalMainOutputs =
-        getInternalMainOutputs(irVertex, irVertexDag, edgeIndexMap, operatorWatermarkManagerMap);
+      final List<NextIntraTaskOperatorInfo> internalMainOutputs;
+      final String key = "main";
```

Comment:
How about moving this into `AdditionalOutputTagProperty` as a public static variable, to ensure there's only a single value for the main output tag? This also has the performance benefit of caching the result of hashCode() and reusing it across all task executions in the executor.

Can you also add a TODO in the comment for the JIRA I've just created? It has become more clear to me after seeing this PR that it may be worthwhile to make the "main" tag more explicit.
https://issues.apache.org/jira/browse/NEMO-348

So I'm thinking something like the following. Let me know if you think this makes sense!
```java
// TODO #348: Consider OutputTagProperty
public static String MAIN_OUTPUT_TAG = "main"
```